src/hotp.rs

```diff
@@ -22,61 +20,79 @@ pub struct HOTP {
     /// The secret key used in the HMAC process.
     ///
     /// Often given as a Base32 key, which can be conveniently initialize using
-    /// the [`HOTP::from_base32`] initializers
+    /// the [`HOTP::from_base32`] constructors.
     secret: Vec<u8>,
+
+    /// The number of digits of the code generated.
+    ///
+    /// This value defaults to 6 if not specified in a constructor.
+    digits: u32,
 }
 
 /// All initializer implementations for the [`HOTP`] struct.
-
 impl HOTP {
     /// Creates a new HOTP instance with a byte-array representation
-    /// of the secret
+    /// of the secret and the number of digits.
     ///
     /// Since only SHA1 was specified in the reference implementation and
-    /// RFC specification, there's no need to initialize with a digest object
-    pub fn new(secret: &[u8]) -> Self {
+    /// RFC specification, there's no need to initialize with a digest object.
+    pub fn new(secret: &[u8], digits: u32) -> Self {
         HOTP {
             secret: secret.to_vec(),
+            digits,
         }
     }
 
-    /// Creates a new HOTP instance from a utf8-encoded string secret
+    /// Creates a new HOTP instance from a utf8-encoded string secret and the number of digits.
+    pub fn new_from_utf8(secret: &str, digits: u32) -> Self {
+        HOTP::new(secret.as_bytes(), digits)
+    }
+
+    /// Creates a new HOTP instance from a base32-encoded string secret and the number of digits.
     ///
-    /// Internally calls [`HOTP::new`] with the string's byte representation
+    /// # Panics
+    /// This method panics if the provided string is not correctly base32 encoded.
+    pub fn new_from_base32(secret: &str, digits: u32) -> Self {
+        let decoded = base32_decode(secret).expect("Failed to decode base32 string");
+        HOTP::new(&decoded, digits)
+    }
+
+    /// Creates a new HOTP instance from a utf8-encoded string secret and a default number of 6 digits.
     pub fn from_utf8(secret: &str) -> Self {
```

Comment:
Can we change the `from_x` constructures into `default_from_x` just for verbosity?